core/internal/filetransfer/file_transfer_manager.go

```diff
@@ -104,15 +104,16 @@ func (fm *fileTransferManager) AddTask(task *Task) {
 }
 
 // completeTask runs the completion callback and updates statistics.
-func (fm *fileTransferManager) completeTask(task *Task) {
-	task.CompletionCallback(task)
+func (fm *fileTransferManager) completeTask(task Task) {
+	taskCompletionCallback := task.GetCompletionCallback()
+	taskCompletionCallback(task)
 
-	if task.Type == UploadTask {
+	if task.GetType() == UploadTask {
 		fm.fileTransferStats.UpdateUploadStats(FileUploadInfo{
-			FileKind:      task.FileKind,
-			Path:          task.Path,
-			UploadedBytes: task.Size,
-			TotalBytes:    task.Size,
+			FileKind:      task.GetFileKind(),
+			Path:          task.GetPath(),
+			UploadedBytes: task.GetSize(),
+			TotalBytes:    task.GetSize(),
```

Comment:
See my other comment about adding a `Complete` method. Then you can do this as part of `Complete` and remove these getters from the `Task` interface.